lib/tools/adb-commands.js

```
@@ -94,4 +98,159 @@ methods.defaultIME = async function () {
+      isKeyboardShown = inputShownMatch[0].split('=')[1] === 'true';
+      let isInputViewShownMatch = /mIsInputViewShown=\w+/gi.exec(stdout);
+      if (isInputViewShownMatch && isInputViewShownMatch[0]) {
+        canCloseKeyboard = isInputViewShownMatch[0].split('=')[1] === 'true';
+      }
+    }
+    return {isKeyboardShown:isKeyboardShown, canCloseKeyboard:canCloseKeyboard};
+  } catch (e) {
+    log.errorAndThrow(`Error finding softkeyboard. Original error: ${e.message}`);
+  }
+};
+
+methods.sendTelnetCommand = async function (command) {
+  log.debug(`Sending telnet command to device: ${command}`);
+  let port = await this.getEmulatorPort();
+  return new Promise(function(resolve, reject) {
+    let conn = net.createConnection(port, 'localhost'),
+        connected = false,
+        readyRegex = /^OK$/m,
+        dataStream = "",
+        res = null;
+    conn.on('connect', function () {
+      log.debug("Socket connection to device created");
+    });
+    conn.on('data', function (data) {
+      data = data.toString('utf8');
+      if (!connected) {
+        if (readyRegex.test(data)) {
+          connected = true;
+          log.debug("Socket connection to device ready");
+          conn.write(command + "\n");
+        }
+      } else {
+        dataStream += data;
+        if (readyRegex.test(data)) {
+          res = dataStream.replace(readyRegex, "").trim();
+          log.debug(`Telnet command got response: ${res}`);
+          conn.write("quit\n");
+        }
+      }
+    });
+    conn.on('close', function () {
+      if (res === null) {
+        reject(new Error("Never got a response from command"));
+      } else {
+        resolve(res);
+      }
+    });
+  });
+};
+
+methods.isAirplaneModeOn = async function () {
+  let stdout = this.shell(['settings', 'get', 'global', 'airplane_mode_on']);
+  return parseInt(stdout) !== 0;
+};
+
+/*
+ * on: 1 (to turn on) or 0 (to turn off)
+ */
+methods.setAirplaneMode = async function (on) {
```

in this and other methods, should we throw an error if `on` is not 1 or 0?